Rework arkyid toxopid

Frolonov opened the pull request:
Unit moving speed & armor improved, also capable of status immunity to survive intense short range combat.
Damage increased to compensate short range, bullet gets armor-Piercing property to handle high tier units.

Arkyid artillery replaced by lightning mount for quick reaction.

Toxopid gets auto tractor beam weapons with freezing to catch & cripple enemy units. 
Shrapnel mount get quicker reload to improved anti-air ability, with life steal frag.
Cannon shell also gets life steal frag to extend combat endurance yet still countered by meltdown & wet+surge.

<img width="326" alt="image" src="https://github.com/user-attachments/assets/cad59123-3892-4907-9154-7a447a7cc02b" />

<img width="442" alt="image" src="https://github.com/user-attachments/assets/45b76490-d4bf-4d41-8dab-c7d1d554bd7e" />

![arkyid1](https://github.com/user-attachments/assets/3785dc33-1286-4c9f-b006-ccfbd4c4a0e6)

![toxo1](https://github.com/user-attachments/assets/19dd3fd0-3c79-4743-8ff3-aa54bef91547)


If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

Frolonov committed:
PR_mindu_combined

PR combined:
scepter, reign
antum, eclipse
quad, oct
vanq, conq
quell, disrupt

Frolonov committed:
include reconstructor PR

Frolonov committed:
include wreck health & damage PR

Frolonov committed:
Rework_corvus3

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
Rework_blasted

Frolonov committed:
balancing tweak

Frolonov committed:
Update Blocks.java

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
blasted rework

Frolonov committed:
Update Blocks.java

Frolonov committed:
Rework_corvus

Frolonov committed:
titan stats update

Frolonov committed:
Rework_arkyid_toxopid

Frolonov committed:
balancing tweak

Frolonov committed:
FX tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
FX tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

coaldeficit commented:
and immediately back to things that should really just be in their own mod right after an almost good pr

Frolonov commented:
At least being a reminder so balancing people know the topic exist. 

Currently there're not that many people post unit rework idea to the point of requiring a dedicated channel.

Frolonov committed:
item crash damage reduction

Frolonov committed:
Rework_arkyid_toxopid

meowkitty9694e commented:
I hate to be rude and all, but i genuinely don't think anyone really wants the vanilla t4 and 5 units to be heavily messed with.

Frolonov commented:
> I hate to be rude and all, but i genuinely don't think anyone really wants the vanilla t4 and 5 units to be heavily messed with.

Well unless there're 500+ downvote, it's hard to tell. There are 10k+ users in Discord & even more outside Discord, yet to see any opinion or specific feedback from that many.

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
Merge branch 'master' into PR_mindu_combined

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
balancing tweak

Frolonov committed:
parallax_tweak

Frolonov committed:
Update TractorBeamTurret.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update TractorBeamTurret.java

Frolonov committed:
Update Blocks.java

Frolonov committed:
Update Blocks.java

Frolonov committed:
Update TractorBeamTurret.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update TractorBeamTurret.java

Frolonov committed:
Update Blocks.java

Frolonov committed:
Update Blocks.java

Frolonov committed:
Update StatusEffects.java

Frolonov committed:
Update Blocks.java

Frolonov committed:
Update TractorBeamTurret.java

Frolonov committed:
Update Units.java

Frolonov committed:
Update TractorBeamTurret.java

Frolonov committed:
Update Blocks.java

Frolonov committed:
Update TractorBeamTurret.java

Frolonov committed:
Update Blocks.java

Frolonov committed:
Update TractorBeamTurret.java

Frolonov committed:
Merge branch 'master' into PR_mindu_combined

Frolonov committed:
Merge pull request #17 from Frolonov/parallax-tweak

Parallax tweak

Frolonov committed:
Merge branch 'master' into blasted-tweak

Frolonov committed:
Merge branch 'PR_mindu_combined' into blasted-tweak

Frolonov committed:
Update Blocks.java

Frolonov committed:
Update gradle.properties

Frolonov committed:
Update gradle.properties

Frolonov committed:
Update gradle.properties

Frolonov committed:
Update gradle.properties

Frolonov committed:
Merge pull request #18 from Frolonov/blasted-tweak

Blasted tweak

Frolonov committed:
Merge branch 'Anuken:master' into parallax-tweak

Frolonov committed:
Merge pull request #20 from Frolonov/parallax-tweak

Parallax tweak

Frolonov committed:
balancing tweak

Frolonov committed:
Update gradle.properties

Frolonov committed:
balancing tweak

Frolonov committed:
Merge branch 'PR_mindu_combined' into Rework-corvus3

Frolonov committed:
Merge pull request #21 from Frolonov/Rework-corvus3

Rework corvus3

Frolonov committed:
Merge branch 'PR_mindu_combined' into Rework-arkyid-toxopid

Frolonov committed:
balancing tweak

BugTD commented:
If Toxopid and Arkyid’s defining weapons and mechanics have been overhauled beyond recognition, why retain names tied to their original roles? This erases their gameplay legacy.

BugTD commented:
Why keep the names Toxopid/Arkyid after stripping their iconic mechanics?

Frolonov commented:
> Why keep the names Toxopid/Arkyid after stripping their iconic mechanics?

Well the sapping beam still there, not sure which part you mean as a total rework.

BugTD commented:
#10580

hydrophobis commented:
Why have you also "rebalanced" multiple turrets, fully reworked the reconstructor build requirements (and made them consume ridiculous amounts of power), changed various sounds for multiple units completely unrelated to the PR, rebalanced unrelated effects, rebalanced explosiveness and fire, reworked certain parts of logic processors, and rebalanced tractor beams?

Frolonov commented:
> Why have you also "rebalanced" multiple turrets, fully reworked the reconstructor build requirements (and made them consume ridiculous amounts of power), changed various sounds for multiple units completely unrelated to the PR, rebalanced unrelated effects, rebalanced explosiveness and fire, reworked certain parts of logic processors, and rebalanced tractor beams?

The PR here only contains changes of 2 units.

As for the combined PR, you could find details regarding each topic here:
https://github.com/Frolonov/Mindustry/pulls

hydrophobis commented:
Not good practice to use the same fork for multiple PRs. In your current state you have (at least) 7 PRs open that make the same changes because you couldn't just wait for your first one to be merged.

Frolonov commented:
> Not good practice to use the same fork for multiple PRs. In your current state you have (at least) 7 PRs open that make the same changes because you couldn't just wait for your first one to be merged.

Not exactly the same since the combined PR only appears in my own repository.
Independent PRs for different units here reduce mixing of comments.
You could actually download JAR of individual PR if you go to Pull request test under each PR & open the 'Upload desktop JAR for testing', which gives the JAR without other units / blocks involved.

Frolonov committed:
Merge branch 'Anuken:master' into Rework-arkyid-toxopid

Anuken commented:
https://github.com/Anuken/Mindustry/pull/10580#issuecomment-5332699666
